extensions/amp-analytics/0.1/events.js

```diff
@@ -1203,6 +1221,27 @@ export class VisibilityTracker extends EventTracker {
     });
   }
 
+  /**
+   * Returns a Promise indicating that we're ready to report the analytics,
+   * in the case of reportWhen: endOfFrame
+   * @return {!Promise}
+   * @private
+   */
+  createReportReadyPromiseForEndOfFrame_() {
+    const deferred = new Deferred();
+    const root = this.root.getRoot();
+    let unloadListener, pageHideListener;
+    root.addEventListener('unload', unloadListener = () => {
+      root.removeEventListener('unload', unloadListener);
+      deferred.resolve();
+    });
+    root.addEventListener('pagehide', pageHideListener = () => {
```

Comment:
will be nice to add comments what those events mean. I know Chrome & Safari handle them differently.

Reply:
Done